src/jdk.compiler/share/classes/com/sun/tools/javac/file/JavacFileManager.java

```diff
@@ -741,13 +741,13 @@ public void close() throws IOException {
     @Override @DefinedBy(Api.COMPILER)
     public ClassLoader getClassLoader(Location location) {
         checkNotModuleOrientedLocation(location);
-        Iterable<? extends File> path = getLocation(location);
-        if (path == null)
+        Collection<? extends Path> paths = getLocationAsPaths(location);
```

Comment:
Minor suggestion:  the use of the simple name `path` or `paths` can be confusing in contexts like this, when the name may refer to an instance of `java.nio.file.Path` or a collection of such items.

You did good by renaming the variable from `path` to `paths`.  A more explicit name would be to use `searchPath` as the name for a series of file system objects.